src/ir/comp.rs

```diff
@@ -343,3 +343,9 @@ impl MutCtx<Instance> for Component {
         self.instances.get_mut(idx)
     }
 }
+
+impl MutCtx<Info> for Component {
```

Comment:
Why do we have to mutate `Info`? My thought for not exposing this method was because once an piece of information is created, it shouldn't really be modified because we no longer remember where it came from

Reply:
The issue is a second pass after all components are created is necessary for this pass to work. We could possibly store an Option<InfoIdx> to a different type of information instead, but the option becomes redundant as it is necessary for all invokes. I also thought about storing a dummy InfoIdx that points to empty, but that requires an extra InfoIdx to be created each time which might be unwanted.

Reply:
The idea is to track the Idxs between the event defined by an invoke + the event defined in the component it corresponds to (and the same with ports). The reason the second pass is necessary is these Idxs are not all generated until all components have been compiled.

Reply:
A large part of the AST functionality will be removed once the IR flow becomes the default so we'll need to define something like that anyways. Also, in astconv, when you're defining the event binds, you should have access to the information about which event is connected to which right?